src/Compilers/Core/Portable/Symbols/LanguageNames.cs

```diff
@@ -28,5 +28,13 @@ public static class LanguageNames
         /// F# is not a supported compile target for the Roslyn compiler.
         /// </remarks>
         public const string FSharp = "F#";
+
+        /// <summary>
+        /// The common name used for the TypeScript language.
+        /// </summary>
+        /// <remarks>
+        /// TypeScript is not a supported compile target for the Roslyn compiler.
+        /// </remarks>
+        public const string TypeScript = "TypeScript";
     }
```

Comment:
Don't add this one. Roslyn does not implement TypeScript language, so it shouldn't be in the public API. Use `InternalLanguageNames.TypeScript` instead.